recipes/pcl/all/patches/0001-flann-target-cmake.patch

```diff
@@ -0,0 +1,298 @@
+--- a/PCLConfig.cmake.in
++++ b/PCLConfig.cmake.in
+@@ -670,8 +670,8 @@ pcl_remove_duplicate_libraries(PCL_COMPONENTS PCL_LIBRARIES)
+ 
+ # Add 3rd party libraries, as user code might include our .HPP implementations
+ list(APPEND PCL_LIBRARIES ${PCL_BOOST_LIBRARIES} ${PCL_OPENNI_LIBRARIES} ${PCL_OPENNI2_LIBRARIES} ${PCL_ENSENSO_LIBRARIES} ${PCL_davidSDK_LIBRARIES} ${PCL_DSSDK_LIBRARIES} ${PCL_RSSDK_LIBRARIES} ${PCL_RSSDK2_LIBRARIES} ${PCL_VTK_LIBRARIES})
+-if (TARGET FLANN::FLANN)
+-  list(APPEND PCL_LIBRARIES FLANN::FLANN)
++if (TARGET flann::flann)
++  list(APPEND PCL_LIBRARIES flann::flann)
+ endif()
+ 
+ if(TARGET QHULL::QHULL)
+diff --git a/cmake/Modules/FindFLANN.cmake b/cmake/Modules/FindFLANN.cmake
+deleted file mode 100644
+index 16ee72fbe..000000000
+--- a/cmake/Modules/FindFLANN.cmake
++++ /dev/null
+@@ -1,223 +0,0 @@
+-#.rst:
+-# FindFLANN
+-# --------
+-#
+-# Try to find FLANN library and headers. This module supports both old released versions
+-# of FLANN ≤ 1.9.1 and newer development versions that ship with a modern config file.
+-#
+-# IMPORTED Targets
+-# ^^^^^^^^^^^^^^^^
+-#
+-# This module defines the :prop_tgt:`IMPORTED` targets:
+-#
+-# ``FLANN::FLANN``
+-#  Defined if the system has FLANN.
+-#
+-# Result Variables
+-# ^^^^^^^^^^^^^^^^
+-#
+-# This module sets the following variables:
+-#
+-# ::
+-#
+-#   FLANN_FOUND               True in case FLANN is found, otherwise false
+-#   FLANN_ROOT                Path to the root of found FLANN installation
+-#
+-# Example usage
+-# ^^^^^^^^^^^^^
+-#
+-# ::
+-#
+-#     find_package(FLANN REQUIRED)
+-#
+-#     add_executable(foo foo.cc)
+-#     target_link_libraries(foo FLANN::FLANN)
+-#
+-
```

Comment:
More of a nitpick, but the patch size could be reduced by wrapping the unnecessary block of code in if(FALSE) ... endif() instead of deleting the file contents.